knockout-pre-rendered.js

```diff
@@ -451,21 +549,38 @@
           fieldValue = value['convert'](fieldValue);
       }
 
-      // Find the field accessor. If the init binding does not point to an observable
-      // or the field parameter doesn't, we try the text and value binding
-      var fieldAccessor = (ko.isObservable(value) ? value : undefined) || 
-                          (isPlainObject(value) ? value['field'] : undefined) ||                             
-                           allBindings.get('text')      ||
-                           allBindings.get('textInput') ||
-                           allBindings.get('value')     ||
-                           allBindings.get('checked')   ||
-                           allBindings.get('html')      ||
-                           allBindings.get('visible')   ||
-                           allBindings.get('enable')    ||
-                           allBindings.get('disable');
-
-      // Finally, update the observable with the value
+    // Look for property writers for explicit fields in the init binding.
+    var initPropertyWriters = allBindings.get('_ko_prerender_initPropertyWriters') || {};
+    // Look for property writers from knockout's built-in two-way bindings
+    var propertyWriters = allBindings.get("_ko_property_writers") || {};
```

Comment:
This also looks like a private field. See my previous question.

Reply:
As with the use of `_twoWayBindings`, this seems to be an undocumented feature that is nonetheless exposed by Knockout, for writers of custom bindings.

Again, from [src/binding/expressionRewriting.js](https://github.com/knockout/knockout/blob/master/src/binding/expressionRewriting.js#L199-L201):
```
// For those developers who rely on _ko_property_writers in their custom bindings, we expose _twoWayBindings as an
// undocumented feature that makes it relatively easy to upgrade to KO 3.0. However, this is still not an official
// public API, and we reserve the right to remove it at any time if we create a real public property writers API.```

Reply:
Okay. The comment implies that `_twoWayBindings` will be the preferred way forward. Is it possible to use `_twoWayBindings` here?

Reply:
`_twoWayBindings` is just a flag (or rather, a collection of flags) instructing Knockout to generate property writer functions for the specified bindings. In other words, if you set `_twoWayBindings.blah = true`, then knockout will generate a `_ko_property_writers` entry for your `blah` binding.  But this really only works for simple value bindings (`something: value`) and not complex bindings (`somthing: { bunch of values... }`).  This is why we can't use it for the `attr` binding, and have to make our own preprocessor, despite it being a knockout built-in binding.

Reply:
In other words, if you are writing a simple binding that looks like `data-bind="blah: someField"`, and you want your binding to be able to write to a non-observable field in the model, then you can set `ko.expressionRewriting._twoWayBindings['blah'] = true;`.  When you do that, the Knockout binding preprocessor will generate an entry in `_ko_property_writers` that looks like: `'blah': function(_z){someField=_z}`.  

And then your binding can write to `someField` even if it's not observable, by doing the following:
```
var writer = allBindings.get("_ko_property_writers")['blah'];
writer(value);
```
This is all that `_twoWayBindings` does.  It tells knockout to generate property writers.

Several of Knockout's built-in bindings (e.g. `textInput`) are already registered in `_twoWayBindings` by default, because they are meant to be bound to user input.

It doesn't bother to do so for `text` or `html` bindings because those are not typically editable by the user, so they don't need to be "writable".

However, when using the `init` binding, we *do* want to write to those properties, given them values extracted from the markup.  

This is why on Line 127, I'm marking the rest of those built-in bindings as `_twoWayBindings`:
```
  ko.utils.extend(ko.expressionRewriting._twoWayBindings, {
    'text': true,
    'html': true,
    'visible': true,
    'enable': true,
    'disable': true
  });
```
Doing so ensures that there will be `_ko_property_writers` generated for them as well, and saves us the trouble of having to make a preprocessor for all of them.